### DISCUSSION TITLE
Experimental Spreadsheet Interface

### DISCUSSION THREAD

**Discussion by JetSetIlly** [robpike/ivy#160] [Show and tell]
I thought it would be interesting to see if Ivy could be made to work with a spreadsheet-like UI. 

I've put together a proof-of-concept application that shows promise. It works for me and I like being able to do native hex/bin arithmetic in a spreadsheet, but I'm not sure if there's any long term mileage in the idea.

I've made it public so people can have a look and maybe build upon the idea.

https://github.com/JetSetIlly/Ivycel

**Comment by fzipp** [comment]
@JetSetIlly It doesn't install due to a replace directive:

```
% go install github.com/jetsetilly/ivycel@latest
go: downloading github.com/jetsetilly/ivycel v0.0.0-20240802203938-8601effa0be0
go: github.com/jetsetilly/ivycel@latest (in github.com/jetsetilly/ivycel@v0.0.0-20240802203938-8601effa0be0):
	The go.mod file for the module providing named packages contains one or
	more replace directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.
```

**Reply by JetSetIlly** [reply]
Yes. I didn't really intend for it to be built with "go install" so I didn't think of that. Thanks for spotting it.

I've updated the go.mod requirements to point to the original repository

Also note that the GUI system uses a C++ library so a C++ compiler and other dev packages will be required. Build requirements for the GUI system are here https://github.com/AllenDang/giu?tab=readme-ov-file#install

**Reply by fzipp** [reply]
Thanks, I was able to install it now.

**Comment by rsc** [comment]
Just seeing this now and wanted to just say: Wow! Very nice!

**Reply by JetSetIlly** [reply]
Thanks Russ!

As I've hopefully made clear this is just a proof of concept sketch but I think it's an idea with promise. I like the idea of a spreadsheet interface for Ivy.

I have some more recent changes to the code that I've not yet pushed to Github. I'll study them to remind myself what I was doing before pushing this weekend.

**Comment by JetSetIlly** [comment]
I've just pushed some local changes that I hadn't committed, for some reason. One of these is the warning in the event of an 'obscured result', as shown in one of the videos.
